src/libsystemd/sd-device/device-util.h

```diff
@@ -38,6 +38,16 @@
              devlink;                                   \
              devlink = sd_device_get_devlink_next(device))
 
+#define FOREACH_DEVICE_CHILD(device, child)                   \
+        for (child = sd_device_get_child_first(device, NULL); \
+             child;                                           \
+             child = sd_device_get_child_next(device, NULL))
+
+#define FOREACH_DEVICE_CHILD_WITH_NAME(device, child, name)     \
```

Comment:
name? it's the syspath, no?

Reply:
or well, it's more than a name, and less than a syspath...

I wonder if this really is so useful? wouldn't most callers either just want the basename of this, or the absolute path, but not this partial path?

Reply:
I do not have good naming for this. Maybe `suffix` for consistency with `sd_device_new_child()`?

- `name` may be a filename of symlink, and different from child's sysname. E.g. SR-IOV: name=virtfn0, but child's sysname is a pci address.
- `name` may be a relative path from the parent's syspath. E.g.
  - parent's syspath: /sys/devices/platform/USBC000:00
  - a child's syspath: /sys/devices/platform/USBC000:00/typec/port0

  then, name=typec/port0

I think providing basename to the caller is not good, as some device has multiple children with the same basename (so, directory is different).
Absolute path may be OK, but redundant, as it always starts with the parent's syspath. I guess callers will always chop the parent's syspath from it.

Reply:
Now, 'name' is renamed to 'suffix'.